pkg/controller/constants/constants.go

```diff
@@ -39,4 +39,14 @@ const (
 
 	// ProvReqAnnotationPrefix is the prefix for annotations that should be pass to ProvisioningRequest as Parameters.
 	ProvReqAnnotationPrefix = "provreq.kueue.x-k8s.io/"
+
+	// WorkloadAnnotation is an annotation set on the Job's PodTemplate to
+	// indicate the name of the admitted Workload corresponding to the Job. The
+	// annotation is set when starting the Job, and removed on stopping the Job.
+	WorkloadAnnotation = "kueue.x-k8s.io/workload"
```

Comment:
What is the difference between this annotation and the Workload name hash?

https://github.com/kubernetes-sigs/kueue/blob/b6222fa0ec020bad0d93be9acdd0305348c4935a/pkg/controller/jobframework/workload_names.go#L35

Reply:
Additionally, is there any case where this annotation and pre-build workload annotation?

Reply:
> What is the difference between this annotation and the Workload name hash?

This new annotation contains the full workload name, for example `job-sample-job-12345`, where 12345 is the hash. The annotation is set on PodTemplate so that newly created Pods carry the information to lookup the Workload.

Reply:
> Additionally, is there any case where this annotation and pre-build workload annotation?

First, note that the two annotations are set at different levels. Prebuilt-workload is the Job-level annotations, whilst the new annotation is set at the PodTemplate level, so that the new Pods "know" their workload.

This might happen on MultiKueue worker cluster. The MK would set the prebuit-workload annotation, and the admission process on the worker will set the new "workload" annotation. However, the new annotation is set on the PodTemplate rather than at the Job level.

Reply:
> This new annotation contains the full workload name, for example job-sample-job-12345, where 12345 is the hash. The annotation is set on PodTemplate so that newly created Pods carry the information to lookup the Workload.

In that case, we can remove the hash mechanism after we graduate this feature to GA?

Reply:
> This might happen on MultiKueue worker cluster. The MK would set the prebuit-workload annotation, and the admission process on the worker will set the new "workload" annotation. However, the new annotation is set on the PodTemplate rather than at the Job level.

I see, so the Pod has only new "workload" annotation. In that case, there seems no to be conflict.
Thanks.